src/tribler/core/config/tribler_config.py

```diff
@@ -152,6 +152,20 @@ def write(self, file: Path = None):
         conf.filename = str(file)
         conf.write()
 
+    def update_from_dict(self, config: Dict):
+        """ Update (patch) current config from dictionary"""
+
+        def update_recursively(settings: BaseSettings, attribute_name: str, attribute_value: Union[Any, Dict]):
+            """ Update setting recursively from dictionary"""
+            if isinstance(attribute_value, dict):
+                for k, v in attribute_value.items():
+                    update_recursively(settings.__getattribute__(attribute_name), k, v)
```

Comment:
**nitpick**: `getattr` should work fine here:
```suggestion
                    update_recursively(getattr(settings, attribute_name), k, v)
```

Reply:
Agree that it works fine here. What are the benefits of using `getattr` over `__getattribute__`?

Reply:
It is more common to use `getattr(a, "b")` instead of `a.__getattr__("b")`; it feels more high-level, and it is shorter and a bit clearer without all these underscores. Just as we usually write `len(collection)` instead of `collection.__len__()`, `iter(sequence)` instead of `sequence.__iter__()`, and `next(iter)` instead of `iter.__next__()`. The longer form with underscores feels a bit like working with private/protected attributes or implementation details, while all of `getattr`/`len`/`iter`/`next` feel more like a public API.

Reply:
Replaced.